### PR TITLE
feat(#130): create events list

### PR DESCRIPTION
**Pull request by lipefree**
Implements #130 and changes the UI a lot to follow to Android colors and components.

### CODE REVIEW

**Comment by lipefree** [comment]
> It's a great PR imo! Is there a reason why the "join" button is not working though? You shouldn't see a join button on an event you joined, you should rather either see "withdraw" or nothing imo

Yeah I was planning to merge last so that I can link everything, it will be working when the PR is merged

File: app/src/main/java/com/github/studydistractor/sdp/eventList/EventListModel.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.github.studydistractor.sdp.eventList
+
+import com.github.studydistractor.sdp.data.Event
+
+interface EventListModel {
+
+    fun getAllEvents(): List<Event>
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/eventList/EventListModel.kt:7]: Comment the model at least

File: app/src/main/java/com/github/studydistractor/sdp/ui/DailyChallengeScreen.kt [hunk 2]
```diff
@@ -11,6 +11,8 @@ import androidx.compose.foundation.verticalScroll
 import androidx.compose.material.Checkbox
 import androidx.compose.material.MaterialTheme
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/DailyChallengeScreen.kt:12]: Update material to material 3 no ?

File: app/src/main/java/com/github/studydistractor/sdp/ui/components/ListedEvent.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            modifier = Modifier
+                .padding(horizontal = 14.dp, vertical = 4.dp)
+                .fillMaxSize(),
+            verticalArrangement = Arrangement.spacedBy(10.dp)
+        ) {
+            Text(
+                text = event.name,
+                style = MaterialTheme.typography.titleLarge,
+                color = MaterialTheme.colorScheme.onPrimaryContainer,
+                modifier = Modifier.testTag("event-list-screen__event-name")
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/components/ListedEvent.kt:55]: use unique tag for testing like adding event id or something.

**Review by bluedrack — CHANGES_REQUESTED** [review]
The pr is good but it will cause a lot of conflict. You probably would have had a better time doing two instead of one to minimize the damage.

File: app/src/main/java/com/github/studydistractor/sdp/eventList/EventListServiceFirebase.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                }
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+                Log.d("Firebase", "loadPost:onCancelled " + error.toException().toString())
+            }
+        })
+    }
+
+    override fun getAllEvents(): List<Event> {
```
[the remark this answers is not in the document]
Comment (lipefree) [reply app/src/main/java/com/github/studydistractor/sdp/eventList/EventListServiceFirebase.kt:37]: I think it is better to not live update the list, for example when the user is scrolling, you don't want to have a changing list.

**Review by AthenaPK — CHANGES_REQUESTED** [review]
Looking good! One comment on the functionality: I think we should only display events that are currently ongoing or are scheduled in the future, it doesn't make sense to include events that happened in the past in the list.

**Comment by lipefree** [comment]
> Looking good! One comment on the functionality: I think we should only display events that are currently ongoing or are scheduled in the future, it doesn't make sense to include events that happened in the past in the list.

The thing is that we made a mistake when implementing the format for dates, but I will look into it before merging

File: app/src/main/java/com/github/studydistractor/sdp/eventList/EventListServiceFirebase.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (bluedrack) [reply app/src/main/java/com/github/studydistractor/sdp/eventList/EventListServiceFirebase.kt:37]: I hadn't seen it that way thanks for clarifying my view of the code!

**Review by aizensoosuke — APPROVED** [review]
LGTM, thanks a lot for all the work :)

**Review by abouquet27 — APPROVED** [review]
LGTM

**Pull request merged**